src/cs/Ssh.Tcp/Services/PortForwardingService.cs

```diff
@@ -506,29 +546,33 @@ public PortForwardingService(SshSession session)
 			{
 				if (request.RequestType == PortForwardRequestType)
 				{
-					int? localPort;
 					try
 					{
-						localPort = await StartForwardingAsync(localEndPoint, cancellation)
+						localPort = await StartForwardingAsync(
+							localIPAddress,
+							(int)portForwardRequest.Port,
+							cancellation)
 							.ConfigureAwait(false);
 					}
 					catch (SocketException)
 					{
 						// The exception is already traced.
-						localPort = null;
 					}
 
 					if (localPort != null)
 					{
+						// The chosen local port may be different from the requested port. Use the
+						// requested port in the response, unless the request was for a random port.
 						response = new PortForwardSuccessMessage
 						{
-							Port = (uint)localPort.Value,
+							Port = portForwardRequest.Port == 0 ? (uint)localPort.Value :
```

Comment:
If `portForwardRequest.Port` is 0, we use the local port where we're forwarding to. What if that local port matches already used other remote port, will that be a conflict?

For example:
1. Host session forwarded from port 5000
2. Client session couldn't start listening on 5000 because it was in use and started listening on another port. 
3. The app blocking port 5000 on the client exits and the port is now available.
4. Host session wants the client to forward on port 0 and client picks 5000.
5. But 5000 is already used by the port forwarded on step 1-2 because it's the remote port.

Reply:
Updated to detect this edge case.